### PR TITLE
Support overriding templates for actions dropdown

### PR DESCRIPTION
**Pull request by Storke**
Related to PR  #3113, it corrects @Soullivaneuh comment

### CODE REVIEW

**Comment by jvasseur** [comment]
This doesn't work because the closing tag can be omited for the li element.

Why not searching for ```<li``` instead ?

**Comment by Soullivaneuh** [comment]
> the closing tag can be omited for the li element.

Why? This **should not** be omited. :-)

> Why not searching for `<li` instead ?

See https://github.com/sonata-project/SonataAdminBundle/pull/3113#issuecomment-125148327.

**Comment by jvasseur** [comment]
Yeah i know it shouldn't really be omitted, but it still is valid html without it so i think it should be supported.

And i've taken into account the discussion in #3113 that's why i said to search for the start of the tag (without the closing >) to include tags that contain attributes.

**Comment by Soullivaneuh** [comment]
@jvasseur Oh didn't understand. You are right.

I would prefer this too. Better, regexp detection?

**Comment by Storke** [comment]
Regexp detection on ```<li``` is a good idea, I do it quickly and rebase my PR.

File: Resources/views/standard_layout.html.twig [hunk 1]
```diff
@@ -261,7 +261,7 @@ file that was distributed with this source code.
 
                                                 {% if _actions|replace({ '<li>': '', '</li>': '' })|trim is not empty %}
                                                     <ul class="nav navbar-nav navbar-right">
-                                                    {% if _actions|split('<li>')|length > 2 %}
+                                                    {% if _actions matches '/(.*<li.*){2}/s' %}
```
Comment (Soullivaneuh) [inline Resources/views/standard_layout.html.twig]: Why `.*` prefix?

And this regexp is too permissive.

Something like `<libutnotli>` will pass AFAIK.

File: Resources/views/standard_layout.html.twig [hunk 2]
```diff
@@ -261,7 +261,8 @@ file that was distributed with this source code.
 
                                                 {% if _actions|replace({ '<li>': '', '</li>': '' })|trim is not empty %}
                                                     <ul class="nav navbar-nav navbar-right">
-                                                    {% if _actions|split('<li>')|length > 2 %}
+                                                    {# Regexp to detect if there is multiple <li tags #}
+                                                    {% if _actions matches '/(.*<li.*){2}/s' %}
```
Comment (Soullivaneuh) [inline Resources/views/standard_layout.html.twig]: Hmm... what the difference?

**Comment by Soullivaneuh** [comment]
I think @Storke was lost on the commits galaxy. :-)

**Comment by Storke** [comment]
Yes, totally lost and i'm not sure that i can revert with this PR...

**Comment by Soullivaneuh** [comment]
Do you still have your modifications on local ? Then redo a commit and push it ton this branch, this should be OK.

**Comment by Storke** [comment]
Ok, I reset my branch HEAD and it will be good for a next commit. 

I put .* to allow any HTML code, your example ```<libutnotli>``` will not be catched because it contains just one ```<li```. But I agree that ```<libutnotli><libutnotli>``` will be catched.

Do you have an idea for a better regexp?

**Comment by Soullivaneuh** [comment]
No time to look at it for the moment but you can play with it here: https://www.debuggex.com/ ;-)

**Comment by jvasseur** [comment]
Something like this <li[ >] could work.

**Comment by Storke** [comment]
This one ````(.*<li[ >].*){2}``` is better, thanks @jvasseur.

But I don't know how to do without ```.*```

**Comment by jvasseur** [comment]
You could do something like this ```<li[ >].*<li[ >]```, but not sure if it is cleaner or not.

**Comment by Soullivaneuh** [comment]
Closes in favor of #3198.

**Pull request closed without merging**